tests/unit_tests/test_matchmaker_algorithm_team_matchmaker.py

```diff
@@ -0,0 +1,248 @@
+import logging
+
+import pytest
+
+from server import config
+from server.matchmaker import CombinedSearch, Search
+from server.matchmaker.algorithm.team_matchmaker import (
+    TeamMatchMaker,
+    UnevenTeamsException
+)
+
+
+@pytest.fixture(scope="module")
+def player_factory(player_factory):
+    player_id_counter = 0
+
+    def make(
+        mean: int = 1500,
+        deviation: int = 500,
+        ladder_games: int = config.NEWBIE_MIN_GAMES+1,
+        name=None
+    ):
+        nonlocal player_id_counter
+        """Make a player with the given ratings"""
+        player = player_factory(
+            ladder_rating=(mean, deviation),
+            ladder_games=ladder_games,
+            login=name,
+            lobby_connection_spec=None,
+            player_id=player_id_counter
+        )
+        player_id_counter += 1
+        return player
+    return make
+
+
+def make_searches(ratings, player_factory):
+    searches = []
+    for i, r in enumerate(ratings):
+        searches.append(Search([player_factory(r + 300, 100, name=f"p{i}")]))
+    return searches
+
+
+def test_team_matchmaker_performance(player_factory, bench, caplog):
+    # Disable debug logging for performance
+    caplog.set_level(logging.INFO)
+    num_searches = 200
+
+    searches = [Search([player_factory(1500, 500, ladder_games=1)]) for _ in range(num_searches)]
+
+    with bench:
+        TeamMatchMaker(4).find(searches)
```

Comment:
This is also including the time it takes to construct the `TeamMatchMaker` object (not that it will be relevant if the number of searches is large enough)